setup.py

```diff
@@ -48,13 +38,10 @@
                                       ]),
       install_requires=[
           'cffi',
-          'msal>=1.16.0,<2',  # http_cache was introduced in MSAL 1.16.0
           'requests>=2.20.0',
+          'azure-identity',
```

Comment:
The question is how user can use your library. Is TokenCredential required? If yes, you need 'azure-identity', if it is optional, then no, you don't need it in the requires.

Reply:
The TokenCredential interface is required. So, leaving this in.

I was thinking of a customer coding a custom class with which satisfies the TokenCredential interface. Python type system allows it via duck typing, but if I follow that logic, any package will never have any dependency. 

I think I over complicated this because the previous code copied the interface of DataLakeCredential..

Reply:
Good point. Yes. User can provide a custom credential that implements TokenCredential protocol. In this case, azure-identity library is not required.

Feel free to remove it from the install_requires.

Reply:
Adding it in extras_require